### DISCUSSION TITLE
LinkNeighborLoader changes the lable of edge_label_index

### DISCUSSION THREAD

**Discussion by SheidaMajouni** [pyg-team/pytorch_geometric#9405] [Q&A] [answered]
Hi,

I’m need help with a bipartite graph I’m working on. based on the nature of my problem, my graph consists of two distinct predefined types of edges: “message passing” and “supervised.” The supervised edges in my problem are already assigned binary labels. I want to use supervised_label = 0 to denote my negative samples. I have carefully prepared my `train_data`, `val_data`, and `test_data`.

I plan to employ `LinkNeighborLoader `to generate batches for my dataset. I understand that due to the nature of neighborhood sampling I can see edges that were not in my `train_data`. However, I’ve encountered an unexpected issue where the labels of some edges are altered, and I’m unsure why this occurs.

I’m focusing on some nodes from my dataset. For my `train_data`, the `edge_label_index `is:

> tensor([[2, 6, 1, 8, 4, 5, 0, 9],
>         [2, 6, 1, 8, 4, 5, 0, 9]])

The corresponding `edge_labels` are:

> tensor([1, 1, 1, 0, 0, 0, 0, 0])

Here, the label for the edge (6, 6) is clearly 1.

However, after executing my code, the `edge_label_index` for `sample_data `becomes:

> tensor([[6, 4, 5, 7, 0, 1, 2, 3],
>         [6, 4, 5, 7, 0, 1, 2, 3]])

With the corresponding edge_labels of:

> tensor([0, 0, 1, 0, 0, 1, 1, 0])

In this case, two new edges (7,7) and (3,3) have been added due to sampling, which is acceptable. However, the label for edge (6,6) has been changed to 0. and why it didn't include edges of (9,9) and (8,8) which are in train edge_label_index?

Could someone clarify if my interpretation is correct? Why would the label change?

```
from torch_geometric.loader import LinkNeighborLoader

# Define seed edges:
edge_label_index = train_data["recipient", "match", "donor"].edge_label_index
print(edge_label_index) # ==> tensor([[2, 6, 1, 8, 4, 5, 0, 9], [2, 6, 1, 8, 4, 5, 0, 9]])

edge_label = train_data["recipient", "match", "donor"].edge_label
print(edge_label) # ==> tensor([1, 1, 1, 0, 0, 0, 0, 0])

train_loader = LinkNeighborLoader(
    data=train_data, 
    num_neighbors=[10],
    edge_label_index= (("recipient", "match", "donor"), edge_label_index),
    edge_label = edge_label,
    batch_size=16, 
    shuffle=True,
)

# Inspect a sample:
sampled_data = next(iter(train_loader))

print(sampled_data["recipient", "match", "donor"].edge_label_index) # ==>tensor([[6, 4, 5, 7, 0, 1, 2, 3], [6, 4, 5, 7, 0, 1, 2, 3]])
print(sampled_data["recipient", "match", "donor"].edge_label) # ==> tensor([0, 0, 1, 0, 0, 1, 1, 0])
```

**Comment by rusty1s** [comment, marked as answer]
Please note that the mini-batch assigns new node IDs based on the sampled subgraph. As such, it is not expected that your `edge_label_index` matches with the original `edge_label_index`. The remapping can be performed via
```
src_n_id = sampled_data['recipient'].n_id
dst_n_id = sampled_data['donor'].n_id
row, col = sampled_data['recipient', 'match', 'donor'].edge_label_index
print(src_n_id[row])
print(dst_n_id[col])

**Reply by SheidaMajouni** [reply]
That is correct! 

Thank you so much! It was a great help. I was so confused.
